pyannote/audio/core/model.py

```diff
@@ -214,7 +213,7 @@ def helper_introspect(
                 frames = self(example_input_array)
             if task is not None:
                 frames = frames[task]
-            _, num_frames, _ = frames.shape
+            num_frames = frames.shape[1]
```

Comment:
I just realized that model output is shaped `(batch_size, num_frames, num_dimension)`.
Shouldn't we update models to output `(batch_size, num_dimension, num_frames)` so that it is coherent with the input `(batch_size, num_channels, num_samples)`?

Reply:
Lets take for example multi-label classification, we're trying to predict animal sounds in audio. Cats, Dogs and Cows. 

For each frame we need to predict what classes are present in it. With the current output, for each frame, we have a predicted set of classes associated with that frame. Isn't that more intuitive than for each class, having a set of frames that belong to that class? With our example that would mean that a frame could belong to multiple classes. This way, the output is more sequential i.e. frame by frame. 

I've never seen a net where the outputs are the second to last dimension, isn't it more common for nets to have the classes output as the final dimension?

Reply:
OK. You convinced me :-) Let's stick to the current output shape.